nixos/modules/system/boot/loader/init-script/init-script.nix

```diff
@@ -22,7 +22,10 @@ in
     boot.loader.initScript = {
 
       enable = mkOption {
-        default = false;
+        # When being booted in as a xen domU
+        # with a provided kernel and initrd,
+        # we need the /sbin/init script.
+        default = config.boot.usePvKernel;
```

Comment:
Every user with PV Kernel would now have to regenerate the manual when you override the default option.
I would rather override this in a module or profile using `mkOverride`.
 We also have `nixos/modules/virtualisation/xen-domU.nix` file, how is this related to your property?

Reply:
I did this here because the same was done for `grub` and `isContainer` in
https://github.com/angerman/nixpkgs/blob/608eaeaeab10593a3f661e9e1012286febbaeb45/nixos/modules/system/boot/loader/grub/grub.nix#L92-L98

The `xen-domU.nix` sets up a xen guest, but expects it to have a boot loader. The situation I am in is in between what xen-domU offers and what the isContainer flags does. However the isContainer flag expects to be a container on a nixOS host.

Reply:
`isContainer` probably was also not the best design decision, because it clutters every module.

Reply:
I’ll need some guidance here. How do I set initScripts.enabled to true if usePvKernel is set to true? Unless the user configured usePvKernel = true; initScripts.enabled = false;

usePvKernel basically implies that initScripts are used as the kernels initrd will very likely expect to find /sbin/init as the standard common entry point for the guest.

Reply:
The key being that nixOS has no control over the kernel or initrd; and the host has no idea about the system and as such treats it as a generic linux.

Reply:
I opened that issue here to discuss the problem: https://github.com/NixOS/nixpkgs/issues/33940

Reply:
We would need for each module where `isContainer` is set a local option to disable this aspect and then set the same option in your pv-kernel module or in a generic container profile. In the linked issue I propose a kernel-less module.

Reply:
That seems to open up another whole can worms. This feels a bit outside of this PR.